sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_models.py

```diff
@@ -229,30 +221,43 @@ def name(self):
     @property
     def properties(self):
         # type: () -> KeyProperties
-        """The key's properties"""
+        """The key's properties
+
+        :rtype: ~azure.keyvault.keys.KeyProperties
+        """
         return self._properties
 
     @property
     def key(self):
         # type: () -> _models.JsonWebKey
-        """The JSON web key"""
+        """The JSON web key
+
+        :rtype: ~azure.keyvault.keys.JsonWebKey
+        """
         return self._key_material
 
     @property
     def key_type(self):
         # type: () -> str
-        """The key's type. See :class:`~azure.keyvault.keys.KeyType` for possible values."""
+        """The key's type. See :class:`~azure.keyvault.keys.KeyType` for possible values.
+
+        :rtype: ~azure.keyvault.keys.KeyType or str
+        """
         return self._key_material.kty
 
     @property
     def key_operations(self):
         # type: () -> list[KeyOperation]
-        """Permitted operations. See :class:`~azure.keyvault.keys.KeyOperation` for possible values."""
+        """Permitted operations. See :class:`~azure.keyvault.keys.KeyOperation` for possible values.
+
+        :rtype: list[~azure.keyvault.keys.KeyOperation or str]
+        """
         return self._key_material.key_ops
 
 
 class DeletedKey(KeyVaultKey):
-    """A deleted key's id, attributes, and cryptographic material, as well as when it will be purged"""
+    """A deleted key's properties and cryptographic material, as well information about its deletion
+    and recovery, if soft-delete is enabled"""
```

Comment:
Phrasing: A deleted key's properties and cryptographic material. If soft-delete is enabled, returns information about its deletion and recovery as well.

Reply:
sweet that sounds better, thanks!